### PR TITLE
make read file line by line utility function

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

File: src/Tomcat/TomcatModel.ts [hunk 1]
```diff
@@ -53,32 +52,25 @@ export class TomcatModel {
...
-                    result = result.concat(line);
-                }
-            });
-            lineReader.on('close', () => {
-                const tmpDirConfiguration: string = result.find((element: string) => {
-                    return element.indexOf(Constants.JAVA_IO_TEMP_DIR_KEY) >= 0;
-                });
-                if (!tmpDirConfiguration || tmpDirConfiguration.length <= 0) {
-                    result = result.concat(`${Constants.JAVA_IO_TEMP_DIR_KEY}=${path.join(catalinaBase, 'temp')}`);
+        const filterFunction: (para: string) => boolean = (para: string): boolean => {
```
Comment (andxu) [inline src/Tomcat/TomcatModel.ts]: (para: string) => boolean = (para: string): boolean => 
what does this mean?

File: src/Utility.ts [hunk 2]
```diff
@@ -115,6 +116,25 @@ export namespace Utility {
         return restartConfig && (httpPort !== newHttpPort || httpsPort !== newHttpsPort);
     }
 
+    export async function readFileLineByLine(file: string, filterFunction?: (value: string) => boolean): Promise<string[]> {
+        let result: string[] = [];
+        await new Promise((resolve: () => void): void => {
```
Comment (andxu) [inline src/Utility.ts]: what does the last void ->  mean

File: src/Utility.ts [hunk 3]
```diff
@@ -115,6 +116,25 @@ export namespace Utility {
         return restartConfig && (httpPort !== newHttpPort || httpsPort !== newHttpsPort);
     }
 
+    export async function readFileLineByLine(file: string, filterFunction?: (value: string) => boolean): Promise<string[]> {
+        let result: string[] = [];
```
Comment (andxu) [inline src/Utility.ts]: let -> const

Comment (SummerSun) [reply src/Utility.ts]: result will be concated with each line ```result = result.concat(line);``` while reading the file

File: src/Tomcat/TomcatModel.ts (hunk 1 above)
Comment (SummerSun) [reply src/Tomcat/TomcatModel.ts]: filterFunction signature

**Pull request merged**